libarchive/archive_entry.c

```diff
@@ -984,7 +984,7 @@ void
 archive_entry_set_ino(struct archive_entry *entry, la_int64_t ino)
 {
 	if (ino < 0) {
-		ino = 0;
+		return;
```

Comment:
I wonder if we should explicit unset the value before returning here?
```
  entry->ae_set &= ~AE_SET_INO;
```

Reply:
makes sense, didn't originally include it to keep it simple assuming it never gets called twice (might be true for `tar` itself, but harder to tell for `libarchive` in general), updated the patch now